include/epoxy/gl.h

```diff
@@ -92,6 +92,11 @@ EPOXY_PUBLIC bool epoxy_has_gl_extension(const char *extension);
 EPOXY_PUBLIC bool epoxy_is_desktop_gl(void);
 EPOXY_PUBLIC int epoxy_gl_version(void);
 
+typedef void * (*epoxy_resolver_failure_handler_t)(const char *name);
```

Comment:
Yes, I have to return something else here.

"glFoo" is really a function pointer named epoxy_glFoo, which initially points to a function named epoxy_glFoo_global_rewrite_ptr. When it is first called, it calls epoxy_glFoo_resolver to GetProcAddress for the symbol, stores that value in epoxy_glFoo, then calls it. This way, on subsequent calls to glFoo, we don't have to look the symbol up again.

We're modifying the bottom of epoxy_glFoo_resolver here, so returning void isn't an option. Neither do I want to return NULL, as whatever function pointer the resolver returns is about to get called.

The X server code for this would look something like:

```c
static void
ZapBadGlxClient(void)
{
    ErrorF("Killing broken GLX client %s\n", GetClientCmdName(currentClient));
    CloseDownClient(currentClient);
}

static void *
GlxResolverFailureHandler(const char *name)
{
    ErrorF("Tried to call unavailable GL function %s\n", name);
    return ZapBadGlxClient;
}
```

Unfortunately this means you only get the diagnostic about which unavailable function was being called on the first resolve failure for it, and subsequent abuse will just hit `ZapBadGlxClient` directly. I think I'm okay with that though, as I don't think jitting a stub function that remembers the name, nor generating dead function stubs in epoxy, would be worth the size or effort.